model_compression_toolkit/core/common/graph/base_node.py

```diff
@@ -540,3 +633,27 @@ def get_simd(self) -> int:
         if _simd <= 0 or int(_simd) != _simd:
             Logger.error(f"SIMD is expected to be a non-positive integer but found: {_simd}")
         return _simd
+
+    def sort_node_candidates(self, fw_info):
+        """
+        Sorts the node candidates.
+        We assume that the candidates are ordered in the following way (for mixed precision purposes):
+            - If the node has a kernel attribute, then we use the kernel weights number of bits to sort the candidates
+            (in descending order). We use the candidate activation number of bits as a secondary order.
+            - If the node doesn't have a kernel we only consider the candidate activation number of bits to sort
+            the candidates in descending order.
+        The operation is done inplace.
+
+        Args:
+            fw_info: FrameworkInfo object about the specific framework (e.g., attributes of different layers' weights to quantize).
+
+        """
+        if self.candidates_quantization_cfg is not None:
+            kernel_attr = fw_info.get_kernel_op_attributes(self.type)[0]
+            if kernel_attr is not None:
```

Comment:
I know that it's considered bad practice to [assign lambda into a variable](https://peps.python.org/pep-0008/#programming-recommendations).
Did you mean that it should be in a nested function instead of lambda? maybe but then I'll need to different nested functions and I don't think it's neater.